### PR TITLE
349 email templates api

### PR DESCRIPTION
**Pull request by gborgonia-ir**
Ticket ID: #349

### CODE REVIEW

**Review by emmaacha — APPROVED** [review]

File: src/controllers/admin/email-template-controller.ts [hunk 1]
```diff
@@ -10,3 +14,180 @@ export const getDefaultEmailTemplate = async (req: Request, res: Response) => {
...
+ * List email templates based on provided filters.
+ * @param req.query.name - Filter by name.
+ * @param req.query.type - Filter by type.
+ * @param req.query.page - Page number for pagination.
+ */
+export const index = async (req: Request, res: Response) => {
+  try {
+    const { name, type, page } = req.query
+
+    const templateType = type === "all" ? "" : type
```
Comment (jatnerubia) [inline src/controllers/admin/email-template-controller.ts:28]: move all logic to service

File: src/controllers/admin/email-template-controller.ts [hunk 2]
```diff
@@ -10,3 +14,180 @@ export const getDefaultEmailTemplate = async (req: Request, res: Response) => {
...
+    const { name, template_type, is_default, subject, content } = req.body
+
+    await addEmailTemplate.validate({
+      name,
+      template_type,
+      subject,
+      content,
+    })
+
+    const emailTemplate = await EmailTemplateService.getById(parseInt(id))
```
Comment (jatnerubia) [inline src/controllers/admin/email-template-controller.ts:144]: same

File: src/controllers/admin/email-template-controller.ts [hunk 3]
```diff
@@ -10,3 +14,180 @@ export const getDefaultEmailTemplate = async (req: Request, res: Response) => {
...
+
+/**
+ * Delete a specific email template by ID.
+ * @param req.params.id - The unique ID of the email template.
+ */
+export const destroy = async (req: Request, res: Response) => {
+  try {
+    const { id } = req.params
+
+    const template = await EmailTemplateService.getById(parseInt(id))
```
Comment (jatnerubia) [inline src/controllers/admin/email-template-controller.ts:177]: same

**Review by jatnerubia — CHANGES_REQUESTED** [review]
just a few changes ^^

**Review by jatnerubia — APPROVED** [review]
Cool

**Pull request merged**